src/main/java/com/google/devtools/build/lib/remote/RemoteAction.java

```diff
@@ -73,6 +77,39 @@ public class RemoteAction {
     this.actionKey = actionKey;
   }
 
+  private void writeDirectoryToFile(FileWriter writer, String root, MerkleTree t) throws IOException {
+    writer.write("[Root proto of " + root + "]\n");
+    if (t != null) {
+      writer.write(t.rootProto.toString());
+      for (String dirName : t.directories.keySet()) {
+        MerkleTree dir = t.directories.get(dirName);
+        writeDirectoryToFile(writer, root + "/" + dirName, dir);
+      }
+    } else {
+      writer.write("[merkleTree is null]\n");
+    }
+  }
+
+  public void writeToFile() {
+    String jsonPath = "/tmp/bazeldebug/" + actionKey.getDigest().getHash() + ".json";
```

Comment:
nit: is it actually json? based on what we write, it seems to me it's not json but maybe I'm confused